### PR TITLE
Check if bundle is done before it could be fetch

### PR DESCRIPTION
**Pull request by janisz**
Fixes: DCOS_OSS-5483

### CODE REVIEW

File: api/rest/cluster_bundle_handler.go [hunk 1]
```diff
@@ -281,19 +281,21 @@ func (c *ClusterBundleHandler) Download(w http.ResponseWriter, r *http.Request)
 	var masterWithBundle node
 	found := false
 	for _, n := range masters {
-		_, err = c.client.Status(ctx, n.baseURL, id)
-		if err != nil {
-			switch err.(type) {
+		bundle, e := c.client.Status(ctx, n.baseURL, id)
```
Comment (makkes) [inline api/rest/cluster_bundle_handler.go]: Is there a specific reason to rename the `err` variable? I'd say it's convenience across our projects to call it `err`.

File: api/rest/bundle_handler.go [hunk 2]
```diff
@@ -235,6 +235,18 @@ func (h BundleHandler) Get(w http.ResponseWriter, r *http.Request) {
...
 	vars := mux.Vars(r)
 	id := vars["id"]
+
+	bundle, err := h.getBundleState(id)
+	if err != nil {
+		writeJSONError(w, http.StatusInternalServerError, err)
+		return
+	}
+	if bundle.Status != Done {
+		writeJSONError(w, http.StatusNotFound,
```
Comment (br-lewis) [inline api/rest/bundle_handler.go]: If we end up with a bundle in the `Failed` state, should that change our error? Is it even possible to get that with cluster bundles?

Comment (janisz) [reply api/rest/bundle_handler.go]: Good point! If bundle is Canceled, Failed or Deleted we should return 404

Comment (janisz) [reply api/rest/bundle_handler.go]: Failed state is here https://github.com/dcos/dcos-diagnostics/pull/150

File: api/rest/cluster_bundle_handler.go (hunk 1 above)
Comment (janisz) [reply api/rest/cluster_bundle_handler.go]:
```api/rest/cluster_bundle_handler.go:284:11: shadow: declaration of "err" shadows declaration at line 273 (govet)
		bundle, err := c.client.Status(ctx, n.baseURL, id)
```

Comment (janisz) [reply api/rest/cluster_bundle_handler.go]: I think it's better to change name than disable linter

File: api/rest/bundle_handler.go (hunk 2 above)
Comment (janisz) [reply api/rest/bundle_handler.go]: Fixed

File: api/rest/cluster_bundle_handler.go (hunk 1 above)
Comment (makkes) [reply api/rest/cluster_bundle_handler.go]: Then I'd prefer sth. more meaningful such as `statusErr`.

Comment (janisz) [reply api/rest/cluster_bundle_handler.go]: Fixed

**Pull request merged**